src/fastest_coeffs.h

```diff
@@ -21,2469 +21,2469 @@ static const struct fastest_coeffs_s
 } fastest_coeffs =
 {	128,
 {
- 8.31472372954840555082e-01,
```

Comment:
Changing these really should not be necessary. These are three large tables that take half as much room as `float` as they would as `double`. Memory space is cheap nowadays, but it almost certainly is worth minimizing memory to CPU bandwidth.

Reply:
I don't understand exactly. This is a float array (`coeff_t coeffs [2464] ;` with `coeff_t == float`)

My only change here is to make the truncation from double to float explicit because the compiler (rightfully) complained that this might be accidental.

Reply:
> complained that this might be accidental.

Its not accidental. The compiler should be informed in a way that doesn't change nearly every line in the file.

Reply:
If a line is like `floatVar = doubleValue` which *does* look accidental then why isn't the correct change `floatVar = floatValue`? And if the whole file consists of such lines then you gotta change the whole file. Nothing wrong with that especially as it can be done with Search&Replace.

The other solution is a suppression via compiler specific pragmas but you objected them in https://github.com/erikd/libsamplerate/pull/82#discussion_r312719227. In the end it doesn't matter which way it is done: Either by passing float values to float variables or by passing double values to float variables and surrounding the region with a pragma saying "don't tell me about truncating conversions". Just needs a decision. See https://github.com/erikd/libsamplerate/pull/82#discussion_r312931906